src/pages/TodoListPage/index.tsx

```diff
@@ -1,3 +1,28 @@
-const TodoListPage = () => <div />;
+import todo from 'apis/todo';
+import NewTodo from 'components/todo/NewTodo';
+import TodoList from 'components/todo/TodoList';
+import { TodosProvider } from 'contexts/todosContext';
+import { useCallback, useEffect, useState } from 'react';
+import type { Todos } from 'types/Todo';
+
+const TodoListPage = () => {
+  const [todos, setTodos] = useState<Todos>();
+
+  const getTodos = useCallback(async () => {
+    const { data } = await todo.getTodos();
+    setTodos(data);
+  }, []);
+
+  useEffect(() => {
+    getTodos();
+  }, [getTodos]);
```

Comment:
getTodos가 다른 곳에서 쓰이지 않으니 그냥 useEffect안에 들어가도 되지 않을까요?

```suggestion
  useEffect(() => {
    const { data } = await todo.getTodos();
    setTodos(data);  
  }, []);
```

Reply:
getTodos가 useEffect안에 들어가면 dependency가 바뀌어서 렌더링이 새로 될 때마다 만들어 질 수 있어서 밖으로 뺐습니다. 물론 현재는 다시 만들어질 일이 없지만 추후에 dependency가 늘어난다면 다시 만들어질 수 있어서 밖으로 뺴는게 좋지 않을까요?

Reply:
useEffect에 콜백함수로 비동기 함수를 주는 것은 지양해야하는 방식이어서 따로 뺐습니다.

Reply:
오 몰랐던 부분인데 알려주셔서 감사합니다 
혹시 관련해서 공식문서나 출처자료를 부탁드려도 될까요?
지양하는 이유나 배경 등에 대해 자세히 알고 싶습니다!

Reply:
useEffect에 들어가는 콜백함수는 undefined나 함수를 리턴해야 하는데 비동기 함수가 들어가면 Promise를 리턴하여 그렇다고 합니다.
https://devtrium.com/posts/async-functions-useeffect